services/ux-backend/handlers/onboarding/peertokens/handler.go

```diff
@@ -1,30 +1,73 @@
 package peertokens
 
 import (
+	"encoding/json"
 	"fmt"
 	"net/http"
+	"strings"
 
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
 	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
 	"github.com/red-hat-storage/ocs-operator/v4/services/ux-backend/handlers"
 
 	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
 	onboardingPrivateKeyFilePath = "/etc/private-key/key"
 )
 
-func HandleMessage(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int) {
+func HandleMessage(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int, cl client.Client) {
 	switch r.Method {
 	case "POST":
-		handlePost(w, r, tokenLifetimeInHours)
+		handlePost(w, r, tokenLifetimeInHours, cl)
 	default:
 		handleUnsupportedMethod(w, r)
 	}
 }
 
-func handlePost(w http.ResponseWriter, _ *http.Request, tokenLifetimeInHours int) {
-	if onboardingToken, err := util.GeneratePeerOnboardingToken(tokenLifetimeInHours, onboardingPrivateKeyFilePath); err != nil {
+func handlePost(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int, cl client.Client) {
+	if r.ContentLength == 0 {
+		http.Error(w, "Request body is empty, expected StorageCluster NamespacedName", http.StatusBadRequest)
+		return
+	}
+
+	var body = struct {
+		StorageClusterNamespacedName string `json:"storageClusterNamespacedName"`
```

Comment:
Why so we need the namespaced name? Don't we just need a namespace?

Reply:
I was thinking from a user perspective, it would make more sense to specify the storageCluster Name they want to onboard to instead of the namespace of the storageCluster. Since storageCluster is a namespace resource, we can have storageClusters with the same name in a different namespace hence chose to go with storageCluster namespacedName to identify the storageCluster user wants to onboard to

Reply:
These endpoints are mainly for the UI to invoke, the UI already has the namespace context and does not need to ask the user for anything. Also, a namespace name as part of a user ask can create mistaken input, where the named resource does not exist in the namespace.